<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Suggested TMT processing pipeline for PNNL DMS *phospho* data.
 • PlexedPiper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Suggested TMT processing pipeline for PNNL DMS *phospho* data.
">
<meta property="og:description" content="PlexedPiper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PlexedPiper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/tmt_pipeline_for_PNNL_DMS_phos.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_for_PNNL_DMS_phos_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_for_PNNL_DMS_phos_pep_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project. Peptide level.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_v1.html">Suggested TMT processing pipeline. Variant 1.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_v1_PNNL_DMS.html">Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
<p>Suggested TMT processing pipeline for PNNL DMS <em>phospho</em> data.</p>
            
      
      
      <div class="hidden name"><code>tmt_pipeline_for_PNNL_DMS_phos.Rmd</code></div>

    </div>

    
    
<p><code>PlexedPiper</code> is the main library for the pipeline. <code>MSnID</code> package is a flexible way of handling MS/MS identifications. It handles MS/MS data filtering part.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PlexedPiper</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PNNL.DMS.utils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MSnID</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="data-source">Data source<a class="anchor" aria-label="anchor" href="#data-source"></a>
</h2>
<p>This vignette demonstrates processing usign PNNL’s DMS as a data source. PNNL DMS is based on MS SQL Server. Typically data is organized into data packages according to experiments. Data package that contains phospho data mirroring <code>PlexedPiperTestData</code> is number <code>3626</code>. Smaller version of the data package, suggested for testing purposes is <code>3625</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_package_num</span> <span class="op">&lt;-</span> <span class="fl">3625</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># if there is no connection to PNNL DMS, vignette is not compiled</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">is_PNNL_DMS_connection_successful</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"There is no connection to PNNL DMS. This code in this vignette won't be evaluated."</span><span class="op">)</span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-msms-ids">Prepare MS/MS IDs<a class="anchor" aria-label="anchor" href="#prepare-msms-ids"></a>
</h2>
<div class="section level3">
<h3 id="read-the-ms-gf-output">Read the MS-GF+ output<a class="anchor" aria-label="anchor" href="#read-the-ms-gf-output"></a>
</h3>
<p>First step is to determine MS-GF+ jobs using the data package number. This simply reads parsed to text output of MS-GF+ search engine. The text files are collated together and the resulting <code>data.frame</code> used to create MSnID object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_msgf_data_from_DMS.html" class="external-link">read_msgf_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 101202 at 61 % FDR</span>
<span class="co">## #peptides: 76398 at 75 % FDR</span>
<span class="co">## #accessions: 78483 at 94 % FDR</span></code></pre>
<p>Phospho dataset involve Ascore jobs for improving phospho-site localization. There should be one Ascore job per data package. The fetched object is a <code>data.frame</code> that links datasets, scans and original PTM localization to newly suggested locations. Importantly it contains <code>AScore</code> column that signifies the confidence of PTM assingment. <code>AScore &gt; 17</code> is considered confident.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ascore</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">get_AScore_results</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/best_PTM_location_by_ascore.html">best_PTM_location_by_ascore</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">ascore</span><span class="op">)</span></code></pre></div>
<p>Remove non-phospho. Need to be sure that phospho symbol is *!</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"grepl(\"\\\\*\", peptide)"</span><span class="op">)</span></code></pre></div>
<p>FDR filter at peptide level.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_msgf_data.html">filter_msgf_data_peptide_level</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="fl">0.01</span><span class="op">)</span></code></pre></div>
<p>Remapping IDs from RefSeq to gene symbols both <code>MSnID</code> object and the FASTA file use for the search.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conv_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/fetch_conversion_table.html" class="external-link">fetch_conversion_table</a></span><span class="op">(</span>organism_name <span class="op">=</span> <span class="st">"Rattus norvegicus"</span>, 
                                  from <span class="op">=</span> <span class="st">"REFSEQ"</span>, to <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span>

<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remap_accessions.html">remap_accessions_refseq_to_gene</a></span><span class="op">(</span><span class="va">msnid</span>,
                                         conversion_table <span class="op">=</span> <span class="va">conv_tbl</span><span class="op">)</span>

<span class="co"># Remove missing accessions</span>
<span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/psms.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/psms.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">accession</span><span class="op">)</span><span class="op">)</span>

<span class="va">fst_dms_pth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">path_to_FASTA_used_by_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">fst_dms_pth_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remap_accessions_fasta.html">remap_accessions_refseq_to_gene_fasta</a></span><span class="op">(</span>
  <span class="va">fst_dms_pth</span>, conversion_table <span class="op">=</span> <span class="va">conv_tbl</span>
<span class="op">)</span></code></pre></div>
<p>Double checking FASTA file. Note, in case of multiple RefSeq mapping to the same gene, we picked the RefSeq entry with the longest sequence. Unfortunately, unlike UniProt, RefSeq does not report, which sequence should be considered as canonical. So to resolve ambiguity we resort to protein length.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span>
<span class="va">fst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">fst_dms_pth_2</span><span class="op">)</span></code></pre></div>
<p>Mapping sites to protein sequence. In a lot of cases it is easier to handle and interpret for example, <code>Hrc-T379t</code>, rather then peptide sequence <code>D.DDDNDGGST*ENVHQAHR.H</code> itself.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/map_mod_sites.html" class="external-link">map_mod_sites</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">fst</span>, 
                       accession_col <span class="op">=</span> <span class="st">"accession"</span>, 
                       peptide_mod_col <span class="op">=</span> <span class="st">"peptide"</span>, 
                       mod_char <span class="op">=</span> <span class="st">"*"</span>,
                       site_delimiter <span class="op">=</span> <span class="st">"lower"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/psms.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                                                 Dataset ResultID  Scan</span>
<span class="co">## 1 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03        1 25367</span>
<span class="co">## 2 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03        5 32449</span>
<span class="co">## 3 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03        6 32449</span>
<span class="co">## 4 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03        7 26428</span>
<span class="co">## 5 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03        8 26428</span>
<span class="co">## 6 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03        9 26428</span>
<span class="co">##   FragMethod SpecIndex Charge PrecursorMZ     DelM DelM_PPM       MH</span>
<span class="co">## 1        HCD         1      4    972.0253 -0.00036  -0.0932 3883.073</span>
<span class="co">## 2        HCD         4      3   1067.8135 -0.00030  -0.0946 3200.422</span>
<span class="co">## 3        HCD         4      3   1067.8135 -0.00030  -0.0946 3200.422</span>
<span class="co">## 4        HCD         5      3   1011.8371 -0.01384  -4.5651 3032.507</span>
<span class="co">## 5        HCD         5      3   1011.8371 -0.01384  -4.5651 3032.507</span>
<span class="co">## 6        HCD         5      3   1011.8371 -0.01384  -4.5651 3032.507</span>
<span class="co">##                       OriginalPeptide        Protein NTT DeNovoScore MSGFScore</span>
<span class="co">## 1 R.RRPT*PATLVLTSDQSSPEVDEDRIPNPLLK.S    NP_073167.1   2         214       191</span>
<span class="co">## 2  R.AGGAES*GPEEDALSLGAEACNFMQPLGGR.S NP_001292067.1   2         183       160</span>
<span class="co">## 3  R.AGGAES*GPEEDALSLGAEACNFMQPLGGR.S NP_001292068.1   2         183       160</span>
<span class="co">## 4    R.VASS*SQELAQVGPAAMLNGQSLQSPPR.A NP_001292067.1   2         151       136</span>
<span class="co">## 5    R.VASS*SQELAQVGPAAMLNGQSLQSPPR.A NP_001292068.1   2         151       136</span>
<span class="co">## 6    R.VASSS*QELAQVGPAAMLNGQSLQSPPR.A NP_001292067.1   2         151       136</span>
<span class="co">##   MSGFDB_SpecEValue Rank_MSGFDB_SpecEValue     EValue QValue PepQValue</span>
<span class="co">## 1        5.6808e-27                      1 1.4918e-19      0         0</span>
<span class="co">## 2        8.7345e-24                      1 2.2851e-16      0         0</span>
<span class="co">## 3        8.7345e-24                      1 2.2851e-16      0         0</span>
<span class="co">## 4        3.5277e-23                      1 9.1588e-16      0         0</span>
<span class="co">## 5        3.5277e-23                      1 9.1588e-16      0         0</span>
<span class="co">## 6        3.5277e-23                      1 9.1588e-16      0         0</span>
<span class="co">##   IsotopeError accession calculatedMassToCharge chargeState</span>
<span class="co">## 1            2   Ppp1r1a               971.5236           4</span>
<span class="co">## 2            1   Synpo2l              1067.4788           3</span>
<span class="co">## 3            1   Synpo2l              1067.4788           3</span>
<span class="co">## 4            1   Synpo2l              1011.5072           3</span>
<span class="co">## 5            1   Synpo2l              1011.5072           3</span>
<span class="co">## 6            1   Synpo2l              1011.5072           3</span>
<span class="co">##   experimentalMassToCharge isDecoy</span>
<span class="co">## 1                 972.0253   FALSE</span>
<span class="co">## 2                1067.8135   FALSE</span>
<span class="co">## 3                1067.8135   FALSE</span>
<span class="co">## 4                1011.8371   FALSE</span>
<span class="co">## 5                1011.8371   FALSE</span>
<span class="co">## 6                1011.8371   FALSE</span>
<span class="co">##                                            spectrumFile spectrumID</span>
<span class="co">## 1 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      25367</span>
<span class="co">## 2 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      32449</span>
<span class="co">## 3 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      32449</span>
<span class="co">## 4 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      26428</span>
<span class="co">## 5 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      26428</span>
<span class="co">## 6 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      26428</span>
<span class="co">##                           pepSeq                             peptide maxAScore</span>
<span class="co">## 1 RRPTPATLVLTSDQSSPEVDEDRIPNPLLK R.RRPTPAT*LVLTSDQSSPEVDEDRIPNPLLK.S      0.00</span>
<span class="co">## 2  AGGAESGPEEDALSLGAEACNFMQPLGGR  R.AGGAES*GPEEDALSLGAEACNFMQPLGGR.S    108.68</span>
<span class="co">## 3  AGGAESGPEEDALSLGAEACNFMQPLGGR  R.AGGAES*GPEEDALSLGAEACNFMQPLGGR.S    108.68</span>
<span class="co">## 4    VASSSQELAQVGPAAMLNGQSLQSPPR    R.VASSS*QELAQVGPAAMLNGQSLQSPPR.A      0.00</span>
<span class="co">## 5    VASSSQELAQVGPAAMLNGQSLQSPPR    R.VASSS*QELAQVGPAAMLNGQSLQSPPR.A      0.00</span>
<span class="co">## 6    VASSSQELAQVGPAAMLNGQSLQSPPR    R.VASSS*QELAQVGPAAMLNGQSLQSPPR.A      0.00</span>
<span class="co">##   msmsScore absParentMassErrorPPM First_AA Last_AA First_AA_First Last_AA_First</span>
<span class="co">## 1       Inf              516.3443       32      61             32            61</span>
<span class="co">## 2       Inf              313.5238      665     693            665           693</span>
<span class="co">## 3       Inf              313.5238      665     693            665           693</span>
<span class="co">## 4       Inf              326.1740      399     425            399           425</span>
<span class="co">## 5       Inf              326.1740      399     425            399           425</span>
<span class="co">## 6       Inf              326.1740      399     425            399           425</span>
<span class="co">##   ProtLen ModShift ModAAs SiteLoc Site SiteCollapsed SiteCollapsedFirst</span>
<span class="co">## 1     171        6      T      38  T38           T38                T38</span>
<span class="co">## 2     978        5      S     670 S670          S670               S670</span>
<span class="co">## 3     978        5      S     670 S670          S670               S670</span>
<span class="co">## 4     978        4      S     403 S403          S403               S403</span>
<span class="co">## 5     978        4      S     403 S403          S403               S403</span>
<span class="co">## 6     978        4      S     403 S403          S403               S403</span>
<span class="co">##          SiteID</span>
<span class="co">## 1  Ppp1r1a-T38t</span>
<span class="co">## 2 Synpo2l-S670s</span>
<span class="co">## 3 Synpo2l-S670s</span>
<span class="co">## 4 Synpo2l-S403s</span>
<span class="co">## 5 Synpo2l-S403s</span>
<span class="co">## 6 Synpo2l-S403s</span></code></pre>
</div>
<div class="section level3">
<h3 id="inference-of-parsimonius-set">Inference of parsimonius set<a class="anchor" aria-label="anchor" href="#inference-of-parsimonius-set"></a>
</h3>
<p>Inference of parsimonius step was skipped in this pipeline. Remapping RefSeq IDs to gene symbols helped the situation of the ambiguous assignment. Demo of inference should be available in other vignettes.</p>
</div>
</div>
<div class="section level2">
<h2 id="prepare-masic-reporter-ion-intensities">Prepare MASIC reporter ion intensities<a class="anchor" aria-label="anchor" href="#prepare-masic-reporter-ion-intensities"></a>
</h2>
<p>Fetching and preparing reporter intensities based on MASIC ouput.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_masic_data_from_DMS.html" class="external-link">read_masic_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span>, 
                                       interference_score <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_masic_data.html">filter_masic_data</a></span><span class="op">(</span><span class="va">masic_data</span>, <span class="fl">0.5</span>, <span class="fl">0</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fetch-study-design-tables">Fetch study design tables<a class="anchor" aria-label="anchor" href="#fetch-study-design-tables"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">study_design_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">get_study_design_by_dataset_package</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">fractions</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">samples</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">references</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-cross-tab">Create cross-tab<a class="anchor" aria-label="anchor" href="#create-cross-tab"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!isDecoy"</span><span class="op">)</span>
<span class="va">aggregation_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SiteID"</span><span class="op">)</span>
<span class="va">quant_cross_tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_crosstab.html">create_crosstab</a></span><span class="op">(</span><span class="va">msnid</span>, 
                                   <span class="va">masic_data</span>, 
                                   <span class="va">aggregation_level</span>, 
                                   <span class="va">fractions</span>, <span class="va">samples</span>, <span class="va">references</span><span class="op">)</span>
<span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 2664   18</span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                        R_01       R_02       R_03        R_04       R_05</span>
<span class="co">## Aak1-S639s      -0.54798756 -0.3528221 -0.7591084 -0.46518317 -0.6804685</span>
<span class="co">## Aak1-S654s      -0.71609119  0.0765027 -0.7579839  0.05178725 -0.1498719</span>
<span class="co">## Aak1-T622t      -0.74928181  0.2854943 -0.8824194 -0.08552446 -0.4203140</span>
<span class="co">## Aak1-T622tS625s -0.33932480 -0.1138501 -0.6839385 -0.29876678 -0.7197100</span>
<span class="co">## Aak1-T622tS626s -0.09656509 -0.1443662 -0.8625578 -0.02124223 -0.4023948</span>
<span class="co">## Aak1-T642t               NA         NA         NA          NA         NA</span>
<span class="co">##                       R_06        R_07       R_08       R_09        S_01</span>
<span class="co">## Aak1-S639s      -0.9119027 -0.92375075 -0.4898818 -0.0768597 -0.58033854</span>
<span class="co">## Aak1-S654s              NA          NA         NA         NA -0.17215866</span>
<span class="co">## Aak1-T622t              NA          NA         NA         NA -0.07470210</span>
<span class="co">## Aak1-T622tS625s         NA          NA         NA         NA -0.23722213</span>
<span class="co">## Aak1-T622tS626s         NA          NA         NA         NA -0.09324999</span>
<span class="co">## Aak1-T642t      -0.9341165 -0.06627768 -0.5125054  0.6196574          NA</span>
<span class="co">##                       S_02        S_03       S_04       S_05        S_06</span>
<span class="co">## Aak1-S639s      -0.7859280 -0.71737909 -1.0244423 -0.2834428 -0.31012621</span>
<span class="co">## Aak1-S654s      -2.3020472 -0.02275759 -0.8836703         NA          NA</span>
<span class="co">## Aak1-T622t      -0.8574086 -0.39572365 -1.3884452         NA          NA</span>
<span class="co">## Aak1-T622tS625s -0.9075300 -0.45022446 -0.9482141         NA          NA</span>
<span class="co">## Aak1-T622tS626s -0.8809523 -0.19529207 -0.9904356         NA          NA</span>
<span class="co">## Aak1-T642t              NA          NA         NA  0.4282869 -0.05693969</span>
<span class="co">##                       S_07       S_08       S_09</span>
<span class="co">## Aak1-S639s      -0.6066893 -0.7573615 -0.6125626</span>
<span class="co">## Aak1-S654s              NA         NA         NA</span>
<span class="co">## Aak1-T622t              NA         NA         NA</span>
<span class="co">## Aak1-T622tS625s         NA         NA         NA</span>
<span class="co">## Aak1-T622tS626s         NA         NA         NA</span>
<span class="co">## Aak1-T642t      -0.2881371 -0.8504565 -0.7601273</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Vladislav Petyuk.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
