<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PlexedPiper">
<title>Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
 • PlexedPiper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
">
<meta property="og:description" content="PlexedPiper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PlexedPiper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos_pep_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project. Peptide level.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_v1.html">Suggested TMT processing pipeline. Variant 1.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_v1_PNNL_DMS.html">Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
</h1>
            
      
      
      <div class="d-none name"><code>tmt_pipeline_for_PNNL_DMS_phos_MoTrPAC.Rmd</code></div>
    </div>

    
    
<p><code>PlexedPiper</code> is the main library for the pipeline. <code>MSnID</code> package is a flexible way of handling MS/MS identifications. It handles MS/MS data filtering part.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PlexedPiper</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PNNL.DMS.utils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MSnID</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="data-source">Data source<a class="anchor" aria-label="anchor" href="#data-source"></a>
</h2>
<p>This vignette demonstrates processing usign PNNL’s DMS as a data source. PNNL DMS is based on MS SQL Server. Typically data is organized into data packages according to experiments. Data package that contains phospho data mirroring <code>PlexedPiperTestData</code> is number <code>3626</code>. Smaller version of the data package, suggested for testing purposes is <code>3625</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_package_num</span> <span class="op">&lt;-</span> <span class="fl">3625</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># if there is no connection to PNNL DMS, vignette is not compiled</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">is_PNNL_DMS_connection_successful</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
   <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"There is no connection to PNNL DMS. This code in this vignette won't be evaluated."</span><span class="op">)</span>
   <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-msms-ids">Prepare MS/MS IDs<a class="anchor" aria-label="anchor" href="#prepare-msms-ids"></a>
</h2>
<div class="section level3">
<h3 id="read-the-ms-gf-output">Read the MS-GF+ output<a class="anchor" aria-label="anchor" href="#read-the-ms-gf-output"></a>
</h3>
<p>First step is to determine MS-GF+ jobs using the data package number. This simply reads parsed to text output of MS-GF+ search engine. The text files are collated together and the resulting <code>data.frame</code> used to create MSnID object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_msgf_data_from_DMS.html" class="external-link">read_msgf_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 101202 at 61 % FDR</span>
<span class="co">## #peptides: 76398 at 75 % FDR</span>
<span class="co">## #accessions: 78483 at 94 % FDR</span></code></pre>
</div>
<div class="section level3">
<h3 id="ascore">AScore<a class="anchor" aria-label="anchor" href="#ascore"></a>
</h3>
<p>Phospho dataset involve Ascore jobs for improving phospho-site localization. There should be one Ascore job per data package. The fetched object is a <code>data.frame</code> that links datasets, scans and original PTM localization to newly suggested locations. Importantly it contains <code>AScore</code> column that signifies the confidence of PTM assingment. <code>AScore &gt; 17</code> is considered confident.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ascore</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">get_AScore_results</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/best_PTM_location_by_ascore.html">best_PTM_location_by_ascore</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">ascore</span><span class="op">)</span></code></pre></div>
<p>Remove non-phospho. Need to be sure that phospho symbol is *!</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"grepl(\"\\\\*\", peptide)"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fdr-filter">FDR filter<a class="anchor" aria-label="anchor" href="#fdr-filter"></a>
</h3>
<p>FDR filter at peptide level.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_msgf_data.html">filter_msgf_data_peptide_level</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="fl">0.01</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="inference-of-parsimonius-set">Inference of parsimonius set<a class="anchor" aria-label="anchor" href="#inference-of-parsimonius-set"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/infer_parsimonious_accessions.html" class="external-link">infer_parsimonious_accessions</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="mapping-sites-to-protein-sequence">Mapping sites to protein sequence<a class="anchor" aria-label="anchor" href="#mapping-sites-to-protein-sequence"></a>
</h3>
<p>Mapping sites to protein sequence. Call creates number of columns describing mapping of the site/s onto the protein sequences. The most important for the user is <code>SiteID</code>.</p>
<p>Prepare FASTA to make sure entry names in FASTA file match <code>MSnID</code> accessions. The plan is to make this conversion automatic. Note, this type of ID extraction will ignore non-RefSeq entries such as contaminants.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fst_dms_pth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">path_to_FASTA_used_by_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">fst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">fst_dms_pth</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">" .*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span><span class="op">)</span> <span class="co"># names must match accessions</span></code></pre></div>
<p>Mapping main call.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/map_mod_sites.html" class="external-link">map_mod_sites</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">fst</span>, 
                       accession_col <span class="op">=</span> <span class="st">"accession"</span>, 
                       peptide_mod_col <span class="op">=</span> <span class="st">"peptide"</span>, 
                       mod_char <span class="op">=</span> <span class="st">"*"</span>,
                       site_delimiter <span class="op">=</span> <span class="st">"lower"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/psms.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                                                 Dataset ResultID  Scan</span>
<span class="co">## 1 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      408 32861</span>
<span class="co">## 2 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      414 31457</span>
<span class="co">## 3 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      636 33482</span>
<span class="co">## 4 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      641 33163</span>
<span class="co">## 5 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      664 33807</span>
<span class="co">## 6 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      701 34776</span>
<span class="co">##   FragMethod SpecIndex Charge PrecursorMZ     DelM DelM_PPM      MH</span>
<span class="co">## 1        HCD        86      3    991.8065 -0.00530  -1.7846 2973.41</span>
<span class="co">## 2        HCD        88      3    991.8090  0.00201   0.6769 2973.41</span>
<span class="co">## 3        HCD       119      3    992.1422 -0.00158  -0.5326 2973.41</span>
<span class="co">## 4        HCD       123      3    992.1429  0.00043   0.1443 2973.41</span>
<span class="co">## 5        HCD       132      3    992.1419 -0.00250  -0.8403 2973.41</span>
<span class="co">## 6        HCD       145      3    992.4756 -0.00463  -1.5576 2973.41</span>
<span class="co">##               OriginalPeptide     Protein NTT DeNovoScore MSGFScore</span>
<span class="co">## 1 R.AAAEGSS*NVFSMFDQTQIQEFK.E NP_036737.1   2         144       114</span>
<span class="co">## 2 R.AAAEGSSNVFS*MFDQTQIQEFK.E NP_036737.1   2         124       108</span>
<span class="co">## 3 R.AAAEGSSNVFS*MFDQTQIQEFK.E NP_036737.1   2         134       112</span>
<span class="co">## 4 R.AAAEGSS*NVFSMFDQTQIQEFK.E NP_036737.1   2         136       110</span>
<span class="co">## 5 R.AAAEGSSNVFS*MFDQTQIQEFK.E NP_036737.1   2         133       101</span>
<span class="co">## 6 R.AAAEGSSNVFS*MFDQTQIQEFK.E NP_036737.1   2         123       104</span>
<span class="co">##   MSGFDB_SpecEValue Rank_MSGFDB_SpecEValue     EValue QValue PepQValue</span>
<span class="co">## 1        1.3776e-17                      1 3.5033e-10      0         0</span>
<span class="co">## 2        1.5700e-17                      1 3.9924e-10      0         0</span>
<span class="co">## 3        3.9576e-17                      1 1.0064e-09      0         0</span>
<span class="co">## 4        4.4128e-17                      1 1.1222e-09      0         0</span>
<span class="co">## 5        7.3846e-17                      1 1.8779e-09      0         0</span>
<span class="co">## 6        1.3331e-16                      1 3.3901e-09      0         0</span>
<span class="co">##   IsotopeError   accession calculatedMassToCharge chargeState</span>
<span class="co">## 1            0 NP_036737.1               991.8083           3</span>
<span class="co">## 2            0 NP_036737.1               991.8083           3</span>
<span class="co">## 3            1 NP_036737.1               991.8083           3</span>
<span class="co">## 4            1 NP_036737.1               991.8083           3</span>
<span class="co">## 5            1 NP_036737.1               991.8083           3</span>
<span class="co">## 6            2 NP_036737.1               991.8083           3</span>
<span class="co">##   experimentalMassToCharge isDecoy</span>
<span class="co">## 1                 991.8065   FALSE</span>
<span class="co">## 2                 991.8090   FALSE</span>
<span class="co">## 3                 992.1422   FALSE</span>
<span class="co">## 4                 992.1429   FALSE</span>
<span class="co">## 5                 992.1419   FALSE</span>
<span class="co">## 6                 992.4756   FALSE</span>
<span class="co">##                                            spectrumFile spectrumID</span>
<span class="co">## 1 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      32861</span>
<span class="co">## 2 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      31457</span>
<span class="co">## 3 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      33482</span>
<span class="co">## 4 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      33163</span>
<span class="co">## 5 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      33807</span>
<span class="co">## 6 MoTrPAC_Pilot_TMT_P_S1_01_DIL_28Oct17_Elm_AQ-17-10-03      34776</span>
<span class="co">##                   pepSeq                     peptide maxAScore msmsScore</span>
<span class="co">## 1 AAAEGSSNVFSMFDQTQIQEFK R.AAAEGSS*NVFSMFDQTQIQEFK.E    17.468       Inf</span>
<span class="co">## 2 AAAEGSSNVFSMFDQTQIQEFK R.AAAEGSS*NVFSMFDQTQIQEFK.E    18.255       Inf</span>
<span class="co">## 3 AAAEGSSNVFSMFDQTQIQEFK R.AAAEGSS*NVFSMFDQTQIQEFK.E    17.468       Inf</span>
<span class="co">## 4 AAAEGSSNVFSMFDQTQIQEFK R.AAAEGSS*NVFSMFDQTQIQEFK.E    17.468       Inf</span>
<span class="co">## 5 AAAEGSSNVFSMFDQTQIQEFK R.AAAEGSS*NVFSMFDQTQIQEFK.E    17.468       Inf</span>
<span class="co">## 6 AAAEGSSNVFSMFDQTQIQEFK R.AAAEGSS*NVFSMFDQTQIQEFK.E    17.468       Inf</span>
<span class="co">##   absParentMassErrorPPM First_AA Last_AA First_AA_First Last_AA_First ProtLen</span>
<span class="co">## 1             1.7627512       10      31             10            31     169</span>
<span class="co">## 2             0.6974017       10      31             10            31     169</span>
<span class="co">## 3           336.6998436       10      31             10            31     169</span>
<span class="co">## 4           337.3753774       10      31             10            31     169</span>
<span class="co">## 5           336.3973658       10      31             10            31     169</span>
<span class="co">## 6           672.8938547       10      31             10            31     169</span>
<span class="co">##   ModShift ModAAs SiteLoc Site SiteCollapsed SiteCollapsedFirst</span>
<span class="co">## 1        6      S      16  S16           S16                S16</span>
<span class="co">## 2        6      S      16  S16           S16                S16</span>
<span class="co">## 3        6      S      16  S16           S16                S16</span>
<span class="co">## 4        6      S      16  S16           S16                S16</span>
<span class="co">## 5        6      S      16  S16           S16                S16</span>
<span class="co">## 6        6      S      16  S16           S16                S16</span>
<span class="co">##             SiteID</span>
<span class="co">## 1 NP_036737.1-S16s</span>
<span class="co">## 2 NP_036737.1-S16s</span>
<span class="co">## 3 NP_036737.1-S16s</span>
<span class="co">## 4 NP_036737.1-S16s</span>
<span class="co">## 5 NP_036737.1-S16s</span>
<span class="co">## 6 NP_036737.1-S16s</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="prepare-masic-reporter-ion-intensities">Prepare MASIC reporter ion intensities<a class="anchor" aria-label="anchor" href="#prepare-masic-reporter-ion-intensities"></a>
</h2>
<p>Fetching and preparing reporter intensities based on MASIC ouput.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_masic_data_from_DMS.html" class="external-link">read_masic_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span>, 
                                       interference_score <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_masic_data.html">filter_masic_data</a></span><span class="op">(</span><span class="va">masic_data</span>, <span class="fl">0.5</span>, <span class="fl">0</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fetch-study-design-tables">Fetch study design tables<a class="anchor" aria-label="anchor" href="#fetch-study-design-tables"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">study_design_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">get_study_design_by_dataset_package</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">fractions</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">samples</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">references</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-cross-tab">Create cross-tab<a class="anchor" aria-label="anchor" href="#create-cross-tab"></a>
</h2>
<p>Aggregation is done directly at <code>SiteID</code> level.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!isDecoy"</span><span class="op">)</span>
<span class="va">aggregation_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SiteID"</span><span class="op">)</span>
<span class="va">quant_cross_tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_crosstab.html">create_crosstab</a></span><span class="op">(</span><span class="va">msnid</span>, 
                                   <span class="va">masic_data</span>, 
                                   <span class="va">aggregation_level</span>, 
                                   <span class="va">fractions</span>, <span class="va">samples</span>, <span class="va">references</span><span class="op">)</span>
<span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 3002   18</span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                            R_01       R_02       R_03       R_04      R_05</span>
<span class="co">## NP_001001514.1-T378t         NA         NA         NA         NA        NA</span>
<span class="co">## NP_001002016.2-S633s         NA         NA         NA         NA        NA</span>
<span class="co">## NP_001002022.1-S232s         NA         NA         NA         NA        NA</span>
<span class="co">## NP_001002277.1-S975s  -0.221334 -0.2668764 -0.4962683 -0.6486341 -0.502687</span>
<span class="co">## NP_001002823.2-S2624s        NA         NA         NA         NA        NA</span>
<span class="co">## NP_001002823.2-S382s         NA         NA         NA         NA        NA</span>
<span class="co">##                             R_06       R_07        R_08        R_09    S_01</span>
<span class="co">## NP_001001514.1-T378t  -0.9102728  0.1452157  0.29764868  1.06356080      NA</span>
<span class="co">## NP_001002016.2-S633s  -0.8376728 -0.1266532 -0.47675809 -0.16171559      NA</span>
<span class="co">## NP_001002022.1-S232s  -0.3585890 -1.1223969 -0.08709753  0.01915121      NA</span>
<span class="co">## NP_001002277.1-S975s  -0.7197738 -0.4621769 -0.57432542 -0.08255740 -1.1161</span>
<span class="co">## NP_001002823.2-S2624s -0.9991695 -0.6964205 -0.69518658 -0.19798040      NA</span>
<span class="co">## NP_001002823.2-S382s  -1.0497954 -0.4427657 -0.57543361 -0.31050097      NA</span>
<span class="co">##                             S_02       S_03      S_04        S_05       S_06</span>
<span class="co">## NP_001001514.1-T378t          NA         NA        NA -0.42961520  0.1019106</span>
<span class="co">## NP_001002016.2-S633s          NA         NA        NA -0.37916148 -0.4258262</span>
<span class="co">## NP_001002022.1-S232s          NA         NA        NA -0.82186420 -0.4086093</span>
<span class="co">## NP_001002277.1-S975s  -0.9010645 -0.4331578 -1.252771 -0.31265712 -0.4093047</span>
<span class="co">## NP_001002823.2-S2624s         NA         NA        NA -0.04835898 -0.6610427</span>
<span class="co">## NP_001002823.2-S382s          NA         NA        NA -0.40651275 -0.5312910</span>
<span class="co">##                             S_07       S_08       S_09</span>
<span class="co">## NP_001001514.1-T378t   0.3944213 -0.7471462  0.4104976</span>
<span class="co">## NP_001002016.2-S633s  -0.3930587 -0.8900900 -0.4564657</span>
<span class="co">## NP_001002022.1-S232s  -0.5548583 -0.8992825 -0.5314916</span>
<span class="co">## NP_001002277.1-S975s  -0.5465060 -0.4536835 -0.6120104</span>
<span class="co">## NP_001002823.2-S2624s -0.6116665 -0.7012145 -0.5859379</span>
<span class="co">## NP_001002823.2-S382s  -0.8688640 -0.7859765 -0.4081825</span></code></pre>
</div>
<div class="section level2">
<h2 id="post-processing--linking-with-ascores">Post-processing. Linking with AScores<a class="anchor" aria-label="anchor" href="#post-processing--linking-with-ascores"></a>
</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x_ascore</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/psms.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">SiteID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>maxAScore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">maxAScore</span><span class="op">)</span><span class="op">)</span>

<span class="va">x</span> <span class="op">&lt;-</span> <span class="va">quant_cross_tab</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"SiteID"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">x_ascore</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>ConfidentSite <span class="op">=</span> <span class="va">maxAScore</span> <span class="op">&gt;</span> <span class="fl">17</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">SiteID</span>, <span class="va">maxAScore</span>, <span class="va">ConfidentSite</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html" class="external-link">everything</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                  SiteID maxAScore ConfidentSite      R_01       R_02       R_03</span>
<span class="co">## 1  NP_001001514.1-T378t    17.980          TRUE        NA         NA         NA</span>
<span class="co">## 2  NP_001002016.2-S633s    58.252          TRUE        NA         NA         NA</span>
<span class="co">## 3  NP_001002022.1-S232s     0.000         FALSE        NA         NA         NA</span>
<span class="co">## 4  NP_001002277.1-S975s    98.625          TRUE -0.221334 -0.2668764 -0.4962683</span>
<span class="co">## 5 NP_001002823.2-S2624s    34.792          TRUE        NA         NA         NA</span>
<span class="co">## 6  NP_001002823.2-S382s    52.139          TRUE        NA         NA         NA</span>
<span class="co">##         R_04      R_05       R_06       R_07        R_08        R_09    S_01</span>
<span class="co">## 1         NA        NA -0.9102728  0.1452157  0.29764868  1.06356080      NA</span>
<span class="co">## 2         NA        NA -0.8376728 -0.1266532 -0.47675809 -0.16171559      NA</span>
<span class="co">## 3         NA        NA -0.3585890 -1.1223969 -0.08709753  0.01915121      NA</span>
<span class="co">## 4 -0.6486341 -0.502687 -0.7197738 -0.4621769 -0.57432542 -0.08255740 -1.1161</span>
<span class="co">## 5         NA        NA -0.9991695 -0.6964205 -0.69518658 -0.19798040      NA</span>
<span class="co">## 6         NA        NA -1.0497954 -0.4427657 -0.57543361 -0.31050097      NA</span>
<span class="co">##         S_02       S_03      S_04        S_05       S_06       S_07       S_08</span>
<span class="co">## 1         NA         NA        NA -0.42961520  0.1019106  0.3944213 -0.7471462</span>
<span class="co">## 2         NA         NA        NA -0.37916148 -0.4258262 -0.3930587 -0.8900900</span>
<span class="co">## 3         NA         NA        NA -0.82186420 -0.4086093 -0.5548583 -0.8992825</span>
<span class="co">## 4 -0.9010645 -0.4331578 -1.252771 -0.31265712 -0.4093047 -0.5465060 -0.4536835</span>
<span class="co">## 5         NA         NA        NA -0.04835898 -0.6610427 -0.6116665 -0.7012145</span>
<span class="co">## 6         NA         NA        NA -0.40651275 -0.5312910 -0.8688640 -0.7859765</span>
<span class="co">##         S_09</span>
<span class="co">## 1  0.4104976</span>
<span class="co">## 2 -0.4564657</span>
<span class="co">## 3 -0.5314916</span>
<span class="co">## 4 -0.6120104</span>
<span class="co">## 5 -0.5859379</span>
<span class="co">## 6 -0.4081825</span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Vladislav Petyuk.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
