<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
 • PlexedPiper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
">
<meta property="og:description" content="PlexedPiper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PlexedPiper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/tmt_pipeline_for_PNNL_DMS_phos.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_for_PNNL_DMS_phos_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_for_PNNL_DMS_phos_pep_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project. Peptide level.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_v1.html">Suggested TMT processing pipeline. Variant 1.
</a>
    </li>
    <li>
      <a href="../articles/tmt_pipeline_v1_PNNL_DMS.html">Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
<p>Suggested TMT processing pipeline. Variant 1. Version for PNNL DMS.</p>
            
      
      
      <div class="hidden name"><code>tmt_pipeline_v1_PNNL_DMS.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PlexedPiper</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PNNL.DMS.utils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MSnID</span><span class="op">)</span></code></pre></div>
<p><code>PlexedPiper</code> is the main library for the pipeline. <code>MSnID</code> package is a flexible way of handling MS/MS identifications. It handles MS/MS data filtering part.</p>
<div class="section level2">
<h2 id="data-source">Data source<a class="anchor" aria-label="anchor" href="#data-source"></a>
</h2>
<p>This pipeline shows how to process TMT data that is processed within PNNL DMS. Data in PNNL DMS is organized using MS SQL Server into data packages. The data package number matching <code>PlexedPiperTestData</code> is 3442. Note, this vignette won’t compile if there is now access to PNNL DMS.</p>
<p>Alternative, smaller variant of 3442 package is 3606.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_package_num</span> <span class="op">&lt;-</span> <span class="fl">3606</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># if there is no connection to PNNL DMS, vignette is not compiled</span>
<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">is_PNNL_DMS_connection_successful</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
   <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"There is no connection to PNNL DMS. This code in this vignette won't be evaluated."</span><span class="op">)</span>
   <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-msms-ids">Prepare MS/MS IDs<a class="anchor" aria-label="anchor" href="#prepare-msms-ids"></a>
</h2>
<div class="section level3">
<h3 id="read-the-ms-gf-output">Read the MS-GF+ output<a class="anchor" aria-label="anchor" href="#read-the-ms-gf-output"></a>
</h3>
<p>Fetching MS/MS search results by data package number. The text files are collated together and the resulting <code>data.frame</code> used to create MSnID object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_msgf_data_from_DMS.html" class="external-link">read_msgf_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 96612 at 30 % FDR</span>
<span class="co">## #peptides: 53098 at 54 % FDR</span>
<span class="co">## #accessions: 72917 at 84 % FDR</span></code></pre>
<p>This <code>MSnID</code> methods corrects for isotope selection error. Occasionally, instrument selects not the lightest or monoisotopic peak, but a peak with +1 or more C13. While MS-GF+ is still capable of correctly identifying those, the downstream calculations of mass measurement error need to be fixed. This method corrects mass measurement errors accounting for selection of other than monoisotopic peak.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/correct_peak_selection.html" class="external-link">correct_peak_selection</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="msms-id-filter-and-peptide-level">MS/MS ID filter and peptide level<a class="anchor" aria-label="anchor" href="#msms-id-filter-and-peptide-level"></a>
</h3>
<p>Main MS/MS filtering step. The <code>filter_msgf_data_peptide_level</code> function wraps-up <code><a href="https://rdrr.io/pkg/MSnID/man/optimize_filter.html" class="external-link">MSnID::optimize_filter</a></code> method and uses <code>PepQValue</code> from MS-GF+ output and absolute deviation of mass measurement error of parent ions (in ppm).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 96612 at 30 % FDR</span>
<span class="co">## #peptides: 53098 at 54 % FDR</span>
<span class="co">## #accessions: 72917 at 84 % FDR</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_msgf_data.html">filter_msgf_data_peptide_level</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="fl">0.01</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 41743 at 0.74 % FDR</span>
<span class="co">## #peptides: 12201 at 0.99 % FDR</span>
<span class="co">## #accessions: 12255 at 3.6 % FDR</span></code></pre>
</div>
<div class="section level3">
<h3 id="switching-annotation-from-refseq-to-gene-symbols">Switching annotation from RefSeq to gene symbols<a class="anchor" aria-label="anchor" href="#switching-annotation-from-refseq-to-gene-symbols"></a>
</h3>
<p>This RefSeq to gene symbols conversion step can be skipped if the downstream analysis relies on RefSeq IDs.</p>
<p>RefSeq is a very redundant database and contains lots of tentative splice isoforms and unconfirmed ORFs. As a result, a lot of peptides (~60%) are shared across multiple RefSeq accessions. This adds to uncertainity which exactly protein we are quantifying. One solution is to switch to less redundant annotations like gene symbol. Moreover gene symbols are more human-interpretable and quite commonly used in downstream analysis.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remap_accessions.html">remap_accessions_refseq_to_gene</a></span><span class="op">(</span><span class="va">msnid</span>, 
                                         organism_name<span class="op">=</span><span class="st">"Rattus norvegicus"</span><span class="op">)</span></code></pre></div>
<p>This step converts sequences names in the fasta file used for MS/MS searches from RefSeq to gene symbols. This is necessary if we are going to count the number of peptide identifications per protein or protein sequence coverage with accessions switched from RefSeq to gene symbols.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path_to_FASTA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">path_to_FASTA_used_by_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">path_to_FASTA_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remap_accessions_fasta.html">remap_accessions_refseq_to_gene_fasta</a></span><span class="op">(</span>
   <span class="va">path_to_FASTA</span>,
   organism_name <span class="op">=</span> <span class="st">"Rattus norvegicus"</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="msms-id-filter-at-protein-level">MS/MS ID filter at protein level<a class="anchor" aria-label="anchor" href="#msms-id-filter-at-protein-level"></a>
</h3>
<p>A while ago proteomics field established hard-and-fast two peptide per protein rule. That is if a protein identified with one peptide it is not considered a confident identification. It works OK, but there is a room for improvement. For example this rule really penalizes short protens. Then there are some very long proteins (e.g. Titin 3.8 MDa) that easily have more then two matching peptides even in reversed sequence. Thus, here we propose to normalize number of peptides per protein length and use that as a filtering criterion.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_num_peptides_per_1000aa.html">compute_num_peptides_per_1000aa</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">path_to_FASTA_gene</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 41743 at 0.74 % FDR</span>
<span class="co">## #peptides: 12201 at 0.99 % FDR</span>
<span class="co">## #accessions: 3876 at 2.7 % FDR</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_msgf_data.html">filter_msgf_data_protein_level</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="fl">0.01</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 32275 at 0.35 % FDR</span>
<span class="co">## #peptides: 9003 at 0.39 % FDR</span>
<span class="co">## #accessions: 2547 at 0.99 % FDR</span></code></pre>
</div>
<div class="section level3">
<h3 id="inference-of-parsimonious-protein-set">Inference of parsimonious protein set<a class="anchor" aria-label="anchor" href="#inference-of-parsimonious-protein-set"></a>
</h3>
<p>The situation when sequence of a certain peptide matches multiple proteins adds complication to the downstream quantitative analysis as it is not clear which protein product this peptide is originating from. There are common ways for selecting a protein set. One is simply retain only uniquely matching peptides (<code>unique_only=TRUE</code>). Shared peptides are simply discarded in this case. Alternatively (in case of <code>unique_only=FALSE</code>) the shared peptides assigned to the proteins with larger number of uniquely mapping peptides. If there is a choice between multiple proteins with equal number of uniquely mapping peptides, the shared peptides is assigned to the first protein according to alphanumeric order.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/infer_parsimonious_accessions.html" class="external-link">infer_parsimonious_accessions</a></span><span class="op">(</span><span class="va">msnid</span>, unique_only<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## MSnID object</span>
<span class="co">## Working directory: "."</span>
<span class="co">## #Spectrum Files:  4 </span>
<span class="co">## #PSMs: 24713 at 0.45 % FDR</span>
<span class="co">## #peptides: 7853 at 0.41 % FDR</span>
<span class="co">## #accessions: 2178 at 0.97 % FDR</span></code></pre>
</div>
<div class="section level3">
<h3 id="remove-decoy-accessions">Remove decoy accessions<a class="anchor" aria-label="anchor" href="#remove-decoy-accessions"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!isDecoy"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="prepare-reporter-ion-intensities">Prepare reporter ion intensities<a class="anchor" aria-label="anchor" href="#prepare-reporter-ion-intensities"></a>
</h2>
<div class="section level3">
<h3 id="read-masic-ouput">Read MASIC ouput<a class="anchor" aria-label="anchor" href="#read-masic-ouput"></a>
</h3>
<p>MASIC is a tool for extracting ion intensities. With proper parameter settings it can be used for extracting TMT (or iTRAQ) reporter ion intensities. In addition it reporter a number of other helpful metrics. Of the most interest are interference score of the parent ion level and S/N at reporter ion level. Interference score reflects what proportion of ion population isolated for fragmentation is due to the targeted ion. The other words 1 - InterferenceScore is due to co-isolated species that have similar elution time and parent ion m/z. S/N at reporter level is the metric that computed by Thermo software.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_masic_data_from_DMS.html" class="external-link">read_masic_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span>, 
                                       interference_score<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                                              Dataset ScanNumber Ion_126.128</span>
<span class="co">## 1: MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02          2    70562.39</span>
<span class="co">## 2: MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02          3    23706.89</span>
<span class="co">## 3: MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02          4    12459.86</span>
<span class="co">## 4: MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02          5        0.00</span>
<span class="co">## 5: MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02          6        0.00</span>
<span class="co">## 6: MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02          8     6166.82</span>
<span class="co">##    Ion_127.125 Ion_127.131 Ion_128.128 Ion_128.134 Ion_129.131 Ion_129.138</span>
<span class="co">## 1:    24864.62    17165.80    35625.00    92236.87     9640.23     8578.05</span>
<span class="co">## 2:    13559.32     5856.83    16322.71    34294.90     4853.11     7938.24</span>
<span class="co">## 3:    11785.91    10932.51    10653.32    12328.62     5959.86     9905.82</span>
<span class="co">## 4:        0.00        0.00        0.00        0.00        0.00        0.00</span>
<span class="co">## 5:    10998.67        0.00    21077.05     2725.50        0.00        0.00</span>
<span class="co">## 6:     1371.27     2418.35     8087.76     5485.35        0.00        0.00</span>
<span class="co">##    Ion_130.135 Ion_130.141 Ion_131.138 Ion_126.128_SignalToNoise</span>
<span class="co">## 1:     6996.69    11833.07    32281.34                     71.47</span>
<span class="co">## 2:        0.00     1465.03    18182.27                     26.12</span>
<span class="co">## 3:     8387.04    11166.70    14053.40                     12.40</span>
<span class="co">## 4:        0.00        0.00        0.00                        NA</span>
<span class="co">## 5:        0.00        0.00     6800.70                        NA</span>
<span class="co">## 6:     1543.48     1943.96     7436.60                      6.92</span>
<span class="co">##    Ion_127.125_SignalToNoise Ion_127.131_SignalToNoise</span>
<span class="co">## 1:                     25.17                     17.38</span>
<span class="co">## 2:                     14.94                      6.45</span>
<span class="co">## 3:                     11.75                     10.90</span>
<span class="co">## 4:                        NA                        NA</span>
<span class="co">## 5:                      9.19                        NA</span>
<span class="co">## 6:                      1.54                      2.71</span>
<span class="co">##    Ion_128.128_SignalToNoise Ion_128.134_SignalToNoise</span>
<span class="co">## 1:                     36.04                     93.32</span>
<span class="co">## 2:                     17.97                     37.77</span>
<span class="co">## 3:                     10.64                     12.31</span>
<span class="co">## 4:                        NA                        NA</span>
<span class="co">## 5:                     17.57                      2.27</span>
<span class="co">## 6:                      9.04                      6.13</span>
<span class="co">##    Ion_129.131_SignalToNoise Ion_129.138_SignalToNoise</span>
<span class="co">## 1:                      9.75                      8.67</span>
<span class="co">## 2:                      5.34                      8.74</span>
<span class="co">## 3:                      5.96                      9.91</span>
<span class="co">## 4:                        NA                        NA</span>
<span class="co">## 5:                        NA                        NA</span>
<span class="co">## 6:                        NA                        NA</span>
<span class="co">##    Ion_130.135_SignalToNoise Ion_130.141_SignalToNoise</span>
<span class="co">## 1:                      7.07                     11.96</span>
<span class="co">## 2:                        NA                      1.61</span>
<span class="co">## 3:                      8.40                     11.18</span>
<span class="co">## 4:                        NA                        NA</span>
<span class="co">## 5:                        NA                        NA</span>
<span class="co">## 6:                      1.72                      2.16</span>
<span class="co">##    Ion_131.138_SignalToNoise InterferenceScore</span>
<span class="co">## 1:                     32.71            0.9961</span>
<span class="co">## 2:                     19.93            0.9925</span>
<span class="co">## 3:                     14.13            1.0000</span>
<span class="co">## 4:                        NA            1.0000</span>
<span class="co">## 5:                      5.66            1.0000</span>
<span class="co">## 6:                      8.26            0.9688</span></code></pre>
</div>
<div class="section level3">
<h3 id="filtering-masic-data">Filtering MASIC data<a class="anchor" aria-label="anchor" href="#filtering-masic-data"></a>
</h3>
<p>In this case (currently recommended filters) we require that at least 50% of ion population is due to the targeted ion and no filter at S/N level.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 133310</span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_masic_data.html">filter_masic_data</a></span><span class="op">(</span><span class="va">masic_data</span>, <span class="fl">0.5</span>, <span class="fl">0</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 127440</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="creating-cross-tab">Creating cross-tab<a class="anchor" aria-label="anchor" href="#creating-cross-tab"></a>
</h2>
<div class="section level3">
<h3 id="laying-out-study-desing">Laying out study desing<a class="anchor" aria-label="anchor" href="#laying-out-study-desing"></a>
</h3>
<p>To convert from PSMs and reporter ion intensities to meaningful quantitative data it is necessary to know what are the samples in the reporter channels and what is the intended reference channel (or combination of channels). The entire study desing captured by three tables - fractions, samples, references.</p>
</div>
<div class="section level3">
<h3 id="fractions">Fractions<a class="anchor" aria-label="anchor" href="#fractions"></a>
</h3>
<p>Dataset to fraction assigment can be read from a separate file or inferred from the dataset names. This example shows the latter.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">masic_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">Dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>PlexID <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"MoTrPAC_Pilot_TMT_W_(S\\d).*"</span>,<span class="st">"\\1"</span>,<span class="va">Dataset</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                                              Dataset PlexID</span>
<span class="co">## 1: MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02     S1</span>
<span class="co">## 2: MoTrPAC_Pilot_TMT_W_S1_02_12Oct17_Elm_AQ-17-09-02     S1</span>
<span class="co">## 3: MoTrPAC_Pilot_TMT_W_S2_01_12Oct17_Elm_AQ-17-09-02     S2</span>
<span class="co">## 4: MoTrPAC_Pilot_TMT_W_S2_02_12Oct17_Elm_AQ-17-09-02     S2</span></code></pre>
</div>
<div class="section level3">
<h3 id="samples">Samples<a class="anchor" aria-label="anchor" href="#samples"></a>
</h3>
<p>This links plex ID, reporter channel with samples. There is a minor difference between ReporterAlias and MeasurementName. MeasurementName is the sample name that will be finally reported. ReporterAlias is an itermediate between ReporterName and MeasurementName and is used for defining what the reference is. There may be a way to query this from DMS. At this point, we’ll read the file from <code>PlexedPiperTestData</code> package.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/study_design/samples.txt"</span>, 
                                package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples</span>,<span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 10 x 5</span></span>
<span class="co">##    PlexID QuantBlock ReporterName ReporterAlias MeasurementName</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> S1              1 126          R_01          R_01           </span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> S1              1 127N         R_03          R_03           </span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> S1              1 127C         R_02          R_02           </span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> S1              1 128N         R_05          R_05           </span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> S1              1 128C         R_04          R_04           </span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> S1              1 129N         S_02          S_02           </span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> S1              1 129C         S_01          S_01           </span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> S1              1 130N         S_04          S_04           </span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> S1              1 130C         S_03          S_03           </span>
<span class="co">## <span style="color: #BCBCBC;">10</span> S1              1 131          ref           <span style="color: #BB0000;">NA</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>Reference can be a certain channel, average of multiple channels or 1. In general form it is an expression with ReporterAlias names as variables. It is evaluated for each PlexID/QuantBlock combination and applied to divide reporter ion intensities within corresponding PlexID/QuantBlock. QuantBlock can be though of as a way of defining sub-plex. In a typical TMT experiment QuantBlock is always 1. In case of 5 pairwise comparisons within TMT10, there will be 5 QuantBlocks (1..5) with reference for each QuantBlock.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">references</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/study_design/references.txt"</span>,
                                   package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">references</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 2 x 3</span></span>
<span class="co">##   PlexID QuantBlock Reference</span>
<span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span>
<span class="co">## <span style="color: #BCBCBC;">1</span> S1              1 ref      </span>
<span class="co">## <span style="color: #BCBCBC;">2</span> S2              1 ref</span></code></pre>
</div>
<div class="section level3">
<h3 id="reading-study-desing-tables-from-dms-data-package-locations">Reading Study Desing Tables from DMS Data Package Locations<a class="anchor" aria-label="anchor" href="#reading-study-desing-tables-from-dms-data-package-locations"></a>
</h3>
<p>By convention, the study design tables should be stored in the data package folder in DMS.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">study_design_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">get_study_design_by_dataset_package</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">fractions</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">samples</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">references</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="creating-quantitative-cross-tab">Creating quantitative cross-tab<a class="anchor" aria-label="anchor" href="#creating-quantitative-cross-tab"></a>
</h3>
<p>Final step when MS/MS IDs and reporter ions linked together, aggregated down to peptide or accession (i.e. protein) level. To retain protein IDs while aggregating to peptide level we suggest <code>aggregation_level &lt;- c("accession","peptide")</code></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aggregation_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"accession"</span><span class="op">)</span>
<span class="va">quant_cross_tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_crosstab.html">create_crosstab</a></span><span class="op">(</span><span class="va">msnid</span>, 
                                   <span class="va">masic_data</span>, 
                                   <span class="va">aggregation_level</span>, 
                                   <span class="va">fractions</span>, <span class="va">samples</span>, <span class="va">references</span><span class="op">)</span>
<span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 2140   18</span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                  R_01        R_02       R_03       R_04        R_05       R_06</span>
<span class="co">## Aar2     -0.425833947 -0.22750410 -0.6327384 -0.4048138 -0.73984597 -0.7172948</span>
<span class="co">## Aars1    -0.008546235  0.52299585 -0.2036391  0.6233634  0.07777239 -0.5622245</span>
<span class="co">## Aarsd1   -0.493124181 -0.35221180 -0.5862425 -0.5514507 -0.38723986 -0.9480688</span>
<span class="co">## Aasdhppt -0.200080510 -0.13412432 -0.4803332 -0.2467851 -0.41344771 -1.1163904</span>
<span class="co">## Abca8a    0.283514199 -0.39489529 -0.4669246  0.4997377 -0.15585177 -0.9529543</span>
<span class="co">## Abce1    -0.203748867 -0.05895448 -0.5425011 -0.2043721 -0.56510223 -0.8850411</span>
<span class="co">##                R_07       R_08        R_09        S_01       S_02       S_03</span>
<span class="co">## Aar2     -0.6163162 -0.4042819  0.02866391 -0.70296182 -0.9735167 -0.6317493</span>
<span class="co">## Aars1    -0.7671264 -0.2315158 -0.07181637  0.57198798  0.3015500 -0.9327534</span>
<span class="co">## Aarsd1   -0.4499616 -0.5698316  0.11402759 -0.37631743 -0.8747977 -0.7770647</span>
<span class="co">## Aasdhppt -0.5757046 -0.4377226 -0.19366971 -0.24615471 -0.9968443 -0.5490088</span>
<span class="co">## Abca8a   -0.4159341 -1.0386196 -0.24260970 -0.76533025 -0.9532367 -0.8305449</span>
<span class="co">## Abce1    -0.5483659 -0.2931637 -0.24382560 -0.09552006 -0.7113775 -0.3476251</span>
<span class="co">##               S_04        S_05       S_06       S_07         S_08       S_09</span>
<span class="co">## Aar2     -1.109744  0.14079541 -0.3316805 -0.2181399 -0.688556180 -0.4159851</span>
<span class="co">## Aars1    -1.998359 -0.51237843 -0.2288511  0.1927945 -0.781111333 -0.2018680</span>
<span class="co">## Aarsd1   -1.018867  0.06778754 -0.2433397 -0.3223110 -0.621658332 -0.5719310</span>
<span class="co">## Aasdhppt -1.176248 -0.20778376 -0.4001400 -0.3037711 -0.638977431 -0.8141141</span>
<span class="co">## Abca8a   -1.202985  0.66608421 -0.6555453 -0.6781002 -0.004690478 -0.8620336</span>
<span class="co">## Abce1    -1.058931 -0.16501937  0.1116517 -0.2789690 -0.573212645 -0.4013663</span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Vladislav Petyuk.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
