<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="A wrapper function for PlexedPiper that performs all the steps
necessary to go from MS-GF+ and MASIC data to Reporter Ion Intensity (RII)
and Ratio Results tables used by the MoTrPAC Bioinformatics Center (BIC)."><title>Wrapper for PlexedPiper — run_plexedpiper • PlexedPiper</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Wrapper for PlexedPiper — run_plexedpiper"><meta property="og:description" content="A wrapper function for PlexedPiper that performs all the steps
necessary to go from MS-GF+ and MASIC data to Reporter Ion Intensity (RII)
and Ratio Results tables used by the MoTrPAC Bioinformatics Center (BIC)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PlexedPiper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos_pep_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project. Peptide level.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_v1.html">Suggested TMT processing pipeline. Variant 1.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_v1_PNNL_DMS.html">Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Wrapper for PlexedPiper</h1>
      
      <div class="d-none name"><code>run_plexedpiper.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A wrapper function for PlexedPiper that performs all the steps
necessary to go from MS-GF+ and MASIC data to Reporter Ion Intensity (RII)
and Ratio Results tables used by the MoTrPAC Bioinformatics Center (BIC).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">run_plexedpiper</span><span class="op">(</span>
  <span class="va">msgf_output_folder</span>,
  <span class="va">fasta_file</span>,
  <span class="va">masic_output_folder</span>,
  ascore_output_folder <span class="op">=</span> <span class="cn">NULL</span>,
  <span class="va">proteomics</span>,
  <span class="va">study_design_folder</span>,
  <span class="va">species</span>,
  <span class="va">annotation</span>,
  file_prefix <span class="op">=</span> <span class="cn">NULL</span>,
  unique_only <span class="op">=</span> <span class="cn">FALSE</span>,
  global_results <span class="op">=</span> <span class="cn">NULL</span>,
  refine_prior <span class="op">=</span> <span class="cn">FALSE</span>,
  write_results_to_file <span class="op">=</span> <span class="cn">TRUE</span>,
  output_folder <span class="op">=</span> <span class="cn">NULL</span>,
  save_env <span class="op">=</span> <span class="cn">FALSE</span>,
  return_results <span class="op">=</span> <span class="cn">FALSE</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>msgf_output_folder</dt>
<dd><p>(character) Path to MSGF+ results folder</p></dd>
<dt>fasta_file</dt>
<dd><p>(character) Path to FASTA file</p></dd>
<dt>masic_output_folder</dt>
<dd><p>(character) MASIC results folder</p></dd>
<dt>ascore_output_folder</dt>
<dd><p>(character) AScore results folder</p></dd>
<dt>proteomics</dt>
<dd><p>(character) Either "pr" - proteomics, "ph" -
phosphoproteomics, "ub" - ubiquitinomics, or "ac" - acetylomics</p></dd>
<dt>study_design_folder</dt>
<dd><p>(character) Folder containing the three study
design tables: fractions.txt, samples.txt, and references.txt</p></dd>
<dt>species</dt>
<dd><p>(character) Scientific name of species (e.g. "Rattus
norvegicus", "Homo sapiens", etc.)</p></dd>
<dt>annotation</dt>
<dd><p>(character) Source for annotations: either <code>RefSeq</code> or
<code>UniProt</code> (case insensitive).</p></dd>
<dt>file_prefix</dt>
<dd><p>(character) Prefix for the file name outputs.</p></dd>
<dt>unique_only</dt>
<dd><p>(logical) Whether to discard peptides that match multiple
proteins in the parsimonious protein inference step. Default <code>FALSE</code>.</p></dd>
<dt>global_results</dt>
<dd><p>(character) Only for PTM experiments. Ratio results
from a global protein abundance experiment. If provided, prioritized
inference will be performed. Otherwise, parsimonious inference is performed
without a prior. See <code><a href="https://rdrr.io/pkg/MSnID/man/infer_parsimonious_accessions.html" class="external-link">infer_parsimonious_accessions</a></code> for
details.</p></dd>
<dt>refine_prior</dt>
<dd><p>(logical) if <code>FALSE</code> (default), peptides are allowed
to match multiple proteins in the prior. That is, the greedy set cover
algorithm is only applied to the set of proteins not in the prior. If
<code>TRUE</code>, the algorithm is applied to the prior and non-prior sets
separately before combining. See
<code><a href="https://rdrr.io/pkg/MSnID/man/infer_parsimonious_accessions.html" class="external-link">infer_parsimonious_accessions</a></code> for more details.</p></dd>
<dt>write_results_to_file</dt>
<dd><p>(logical) Whether to write the results to files.</p></dd>
<dt>output_folder</dt>
<dd><p>(character) Output folder name to save results. If not
provided, it will save it to the current directory.</p></dd>
<dt>save_env</dt>
<dd><p>(logical) Whether to save the R environment to the output
folder.</p></dd>
<dt>return_results</dt>
<dd><p>(logical) Whether to return the ratio and rii results.</p></dd>
<dt>verbose</dt>
<dd><p>(logical) Whether to show messages describing steps.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>(list) If <code>return_results</code> is <code>TRUE</code>, it returns list with ratio and
RII data frames.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">run_plexedpiper</span><span class="op">(</span>msgf_output_folder <span class="op">=</span> <span class="st">"~/path/to/msgfplus/"</span>,</span>
<span class="r-in">                           fasta_file  <span class="op">=</span> <span class="st">"~/path/to/fasta/sequence.fasta"</span>,</span>
<span class="r-in">                           masic_output_folder <span class="op">=</span> <span class="st">"~/path/to/masic-results/"</span>,</span>
<span class="r-in">                           ascore_output_folder <span class="op">=</span> <span class="st">"~/path/to/ascore-results/"</span>,</span>
<span class="r-in">                           proteomics <span class="op">=</span> <span class="st">"ph"</span>,</span>
<span class="r-in">                           study_design_folder <span class="op">=</span> <span class="st">"~/path/to/study-design/"</span>,</span>
<span class="r-in">                           species <span class="op">=</span> <span class="st">"Rattus norvegicus"</span>,</span>
<span class="r-in">                           annotation <span class="op">=</span> <span class="st">"RefSeq"</span>,</span>
<span class="r-in">                           global_results <span class="op">=</span> <span class="st">"~/path/to/global/ratio.txt"</span>,</span>
<span class="r-in">                           unique_only <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">                           refine_prior <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">                           output_folder <span class="op">=</span> <span class="st">"~/path/to/pp-results/"</span>,</span>
<span class="r-in">                           file_prefix <span class="op">=</span> <span class="st">"msgfplus-pp-results"</span>,</span>
<span class="r-in">                           return_results <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span class="r-in">                           verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Vladislav Petyuk.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

