<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PlexedPiper">
<title>Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
 • PlexedPiper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
">
<meta property="og:description" content="PlexedPiper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PlexedPiper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_for_PNNL_DMS_phos_pep_MoTrPAC.html">Suggested TMT processing pipeline for PNNL DMS *phospho* data
for MoTrPAC project. Peptide level.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_v1.html">Suggested TMT processing pipeline. Variant 1.
</a>
    <a class="dropdown-item" href="../articles/tmt_pipeline_v1_PNNL_DMS.html">Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PNNL-Comp-Mass-Spec/PlexedPiper/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Suggested TMT processing pipeline. Variant 1.
Version for PNNL DMS.
</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PNNL-Comp-Mass-Spec/PlexedPiper/blob/HEAD/vignettes/tmt_pipeline_v1_PNNL_DMS.Rmd" class="external-link"><code>vignettes/tmt_pipeline_v1_PNNL_DMS.Rmd</code></a></small>
      <div class="d-none name"><code>tmt_pipeline_v1_PNNL_DMS.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pnnl-comp-mass-spec.github.io/PlexedPiper/">PlexedPiper</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PNNL.DMS.utils</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MSnID</span><span class="op">)</span></span></code></pre></div>
<p><code>PlexedPiper</code> is the main library for the pipeline.
<code>MSnID</code> package is a flexible way of handling MS/MS
identifications. It handles MS/MS data filtering part.</p>
<div class="section level2">
<h2 id="data-source">Data source<a class="anchor" aria-label="anchor" href="#data-source"></a>
</h2>
<p>This pipeline shows how to process TMT data that is processed within
PNNL DMS. Data in PNNL DMS is organized using MS SQL Server into data
packages. The data package number matching
<code>PlexedPiperTestData</code> is 3442. Note, this vignette won’t
compile if there is now access to PNNL DMS.</p>
<p>Alternative, smaller variant of 3442 package is 3606.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_package_num</span> <span class="op">&lt;-</span> <span class="fl">3606</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if there is no connection to PNNL DMS, vignette is not compiled</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">is_PNNL_DMS_connection_successful</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"There is no connection to PNNL DMS. This code in this vignette won't be evaluated."</span><span class="op">)</span></span>
<span>   <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-msms-ids">Prepare MS/MS IDs<a class="anchor" aria-label="anchor" href="#prepare-msms-ids"></a>
</h2>
<div class="section level3">
<h3 id="read-the-ms-gf-output">Read the MS-GF+ output<a class="anchor" aria-label="anchor" href="#read-the-ms-gf-output"></a>
</h3>
<p>Fetching MS/MS search results by data package number. The text files
are collated together and the resulting <code>data.frame</code> used to
create MSnID object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_msgf_data_from_DMS.html" class="external-link">read_msgf_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
<p>This <code>MSnID</code> methods corrects for isotope selection error.
Occasionally, instrument selects not the lightest or monoisotopic peak,
but a peak with +1 or more C13. While MS-GF+ is still capable of
correctly identifying those, the downstream calculations of mass
measurement error need to be fixed. This method corrects mass
measurement errors accounting for selection of other than monoisotopic
peak.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/correct_peak_selection.html" class="external-link">correct_peak_selection</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="msms-id-filter-and-peptide-level">MS/MS ID filter and peptide level<a class="anchor" aria-label="anchor" href="#msms-id-filter-and-peptide-level"></a>
</h3>
<p>Main MS/MS filtering step. The
<code>filter_msgf_data_peptide_level</code> function wraps-up
<code><a href="https://rdrr.io/pkg/MSnID/man/optimize_filter.html" class="external-link">MSnID::optimize_filter</a></code> method and uses
<code>PepQValue</code> from MS-GF+ output and absolute deviation of mass
measurement error of parent ions (in ppm).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span>
<span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_msgf_data.html">filter_msgf_data_peptide_level</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="switching-annotation-from-refseq-to-gene-symbols">Switching annotation from RefSeq to gene symbols<a class="anchor" aria-label="anchor" href="#switching-annotation-from-refseq-to-gene-symbols"></a>
</h3>
<p>This RefSeq to gene symbols conversion step can be skipped if the
downstream analysis relies on RefSeq IDs.</p>
<p>RefSeq is a very redundant database and contains lots of tentative
splice isoforms and unconfirmed ORFs. As a result, a lot of peptides
(~60%) are shared across multiple RefSeq accessions. This adds to
uncertainity which exactly protein we are quantifying. One solution is
to switch to less redundant annotations like gene symbol. Moreover gene
symbols are more human-interpretable and quite commonly used in
downstream analysis.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remap_accessions.html">remap_accessions_refseq_to_gene</a></span><span class="op">(</span><span class="va">msnid</span>, </span>
<span>                                         organism_name<span class="op">=</span><span class="st">"Rattus norvegicus"</span><span class="op">)</span></span></code></pre></div>
<p>This step converts sequences names in the fasta file used for MS/MS
searches from RefSeq to gene symbols. This is necessary if we are going
to count the number of peptide identifications per protein or protein
sequence coverage with accessions switched from RefSeq to gene
symbols.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path_to_FASTA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html" class="external-link">path_to_FASTA_used_by_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span></span>
<span><span class="va">path_to_FASTA_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remap_accessions_fasta.html">remap_accessions_refseq_to_gene_fasta</a></span><span class="op">(</span></span>
<span>   <span class="va">path_to_FASTA</span>,</span>
<span>   organism_name <span class="op">=</span> <span class="st">"Rattus norvegicus"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="msms-id-filter-at-protein-level">MS/MS ID filter at protein level<a class="anchor" aria-label="anchor" href="#msms-id-filter-at-protein-level"></a>
</h3>
<p>A while ago proteomics field established hard-and-fast two peptide
per protein rule. That is if a protein identified with one peptide it is
not considered a confident identification. It works OK, but there is a
room for improvement. For example this rule really penalizes short
protens. Then there are some very long proteins (e.g. Titin 3.8 MDa)
that easily have more then two matching peptides even in reversed
sequence. Thus, here we propose to normalize number of peptides per
protein length and use that as a filtering criterion.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_num_peptides_per_1000aa.html">compute_num_peptides_per_1000aa</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">path_to_FASTA_gene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span>
<span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_msgf_data.html">filter_msgf_data_protein_level</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="inference-of-parsimonious-protein-set">Inference of parsimonious protein set<a class="anchor" aria-label="anchor" href="#inference-of-parsimonious-protein-set"></a>
</h3>
<p>The situation when sequence of a certain peptide matches multiple
proteins adds complication to the downstream quantitative analysis as it
is not clear which protein product this peptide is originating from.
There are common ways for selecting a protein set. One is simply retain
only uniquely matching peptides (<code>unique_only=TRUE</code>). Shared
peptides are simply discarded in this case. Alternatively (in case of
<code>unique_only=FALSE</code>) the shared peptides assigned to the
proteins with larger number of uniquely mapping peptides. If there is a
choice between multiple proteins with equal number of uniquely mapping
peptides, the shared peptides is assigned to the first protein according
to alphanumeric order.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/infer_parsimonious_accessions.html" class="external-link">infer_parsimonious_accessions</a></span><span class="op">(</span><span class="va">msnid</span>, unique_only<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remove-decoy-accessions">Remove decoy accessions<a class="anchor" aria-label="anchor" href="#remove-decoy-accessions"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!isDecoy"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="prepare-reporter-ion-intensities">Prepare reporter ion intensities<a class="anchor" aria-label="anchor" href="#prepare-reporter-ion-intensities"></a>
</h2>
<div class="section level3">
<h3 id="read-masic-ouput">Read MASIC ouput<a class="anchor" aria-label="anchor" href="#read-masic-ouput"></a>
</h3>
<p>MASIC is a tool for extracting ion intensities. With proper parameter
settings it can be used for extracting TMT (or iTRAQ) reporter ion
intensities. In addition it reporter a number of other helpful metrics.
Of the most interest are interference score of the parent ion level and
S/N at reporter ion level. Interference score reflects what proportion
of ion population isolated for fragmentation is due to the targeted ion.
The other words 1 - InterferenceScore is due to co-isolated species that
have similar elution time and parent ion m/z. S/N at reporter level is
the metric that computed by Thermo software.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_masic_data_from_DMS.html" class="external-link">read_masic_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span>, </span>
<span>                                       interference_score<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering-masic-data">Filtering MASIC data<a class="anchor" aria-label="anchor" href="#filtering-masic-data"></a>
</h3>
<p>In this case (currently recommended filters) we require that at least
50% of ion population is due to the targeted ion and no filter at S/N
level.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">)</span></span>
<span><span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_masic_data.html">filter_masic_data</a></span><span class="op">(</span><span class="va">masic_data</span>, <span class="fl">0.5</span>, <span class="fl">0</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="creating-cross-tab">Creating cross-tab<a class="anchor" aria-label="anchor" href="#creating-cross-tab"></a>
</h2>
<div class="section level3">
<h3 id="laying-out-study-desing">Laying out study desing<a class="anchor" aria-label="anchor" href="#laying-out-study-desing"></a>
</h3>
<p>To convert from PSMs and reporter ion intensities to meaningful
quantitative data it is necessary to know what are the samples in the
reporter channels and what is the intended reference channel (or
combination of channels). The entire study desing captured by three
tables - fractions, samples, references.</p>
</div>
<div class="section level3">
<h3 id="fractions">Fractions<a class="anchor" aria-label="anchor" href="#fractions"></a>
</h3>
<p>Dataset to fraction assigment can be read from a separate file or
inferred from the dataset names. This example shows the latter.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">masic_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">Dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>PlexID <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"MoTrPAC_Pilot_TMT_W_(S\\d).*"</span>,<span class="st">"\\1"</span>,<span class="va">Dataset</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="samples">Samples<a class="anchor" aria-label="anchor" href="#samples"></a>
</h3>
<p>This links plex ID, reporter channel with samples. There is a minor
difference between ReporterAlias and MeasurementName. MeasurementName is
the sample name that will be finally reported. ReporterAlias is an
itermediate between ReporterName and MeasurementName and is used for
defining what the reference is. There may be a way to query this from
DMS. At this point, we’ll read the file from
<code>PlexedPiperTestData</code> package.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/study_design/samples.txt"</span>, </span>
<span>                                package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples</span>,<span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>Reference can be a certain channel, average of multiple channels or
1. In general form it is an expression with ReporterAlias names as
variables. It is evaluated for each PlexID/QuantBlock combination and
applied to divide reporter ion intensities within corresponding
PlexID/QuantBlock. QuantBlock can be though of as a way of defining
sub-plex. In a typical TMT experiment QuantBlock is always 1. In case of
5 pairwise comparisons within TMT10, there will be 5 QuantBlocks (1..5)
with reference for each QuantBlock.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">references</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/study_design/references.txt"</span>,</span>
<span>                                   package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">references</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="reading-study-desing-tables-from-dms-data-package-locations">Reading Study Desing Tables from DMS Data Package Locations<a class="anchor" aria-label="anchor" href="#reading-study-desing-tables-from-dms-data-package-locations"></a>
</h3>
<p>By convention, the study design tables should be stored in the data
package folder in DMS.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_design_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_study_design.html" class="external-link">read_study_design_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span></span>
<span><span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">fractions</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">samples</span></span>
<span><span class="va">references</span> <span class="op">&lt;-</span> <span class="va">study_design_tables</span><span class="op">$</span><span class="va">references</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="creating-quantitative-cross-tab">Creating quantitative cross-tab<a class="anchor" aria-label="anchor" href="#creating-quantitative-cross-tab"></a>
</h3>
<p>Final step when MS/MS IDs and reporter ions linked together,
aggregated down to peptide or accession (i.e. protein) level. To retain
protein IDs while aggregating to peptide level we suggest
<code>aggregation_level &lt;- c("accession","peptide")</code></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aggregation_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"accession"</span><span class="op">)</span></span>
<span><span class="va">quant_cross_tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_crosstab.html">create_crosstab</a></span><span class="op">(</span><span class="va">msnid</span>, </span>
<span>                                   <span class="va">masic_data</span>, </span>
<span>                                   <span class="va">aggregation_level</span>, </span>
<span>                                   <span class="va">fractions</span>, <span class="va">samples</span>, <span class="va">references</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">quant_cross_tab</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Vladislav Petyuk.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
